api/rbdigital.py

```diff
@@ -1316,6 +1288,106 @@ def get_delta(self, from_date=None, to_date=None, verbosity=None):
         response = self.request(url, params=args, verbosity=verbosity)
         return response.json()
 
+    def align_dates_to_available_snapshots(self, from_date=None, to_date=None):
+        """
+        Given specified begin and end dates for a delta, return the best dates from those available.
+
+        Note: It might be useful to raise an exception or log a message if either of the
+        "best" dates is too distant from the associated specified date.
+
+        The endpoint utilized returns a JSON array of "snapshot" objects (nb: tenantId is the library ID):
+            Example snapshot format:
+                "tenantId": 525,
+                "asOf": "2020-04-07",
+                "eBookCount": 1630,
+                "eAudioCount": 13414,
+                "totalCount": 15044
+
+        :return Best available begin and end dates.
+        """
+        SNAPSHOT_DATE_FORMAT = "%Y-%m-%d"
+        # fuzzy_binary_search relative match values
+        INDEXED_EQUALS_MATCH = 0
+        INDEXED_GREATER_THAN_MATCH = -1
+        INDEXED_LESS_THAN_MATCH = 1
+
+        url = "%s/libraries/%s/book-holdings/delta/available-dates" % (self.base_url, str(self.library_id))
+
+        response = self.request(url)
+        try:
+            snapshots = response.json()
+        except Exception as e:
```

Comment:
I'm wary of any kind of blanket `except`; I think this should just be ValueError.

Reply:
Yeah, that makes sense. I should add a test for that, too.